### ISSUE TITLE
Implement --interface option for non-linux systems

### ISSUE THREAD

**Issue by Arriven** [Arriven/db1000n#529]
--interface option uses syscall.BindToDevice which is only available for linux. It should be possible to use net.InterfaceByName inside conn.Control to get an addr that could then be transformed into sockaddr for regular syscall.Bind which is available for all unix systems. Doing it in the control function should be enough to not have troubles when network changes at runtime

**Comment by deputinizer** [comment]
Looks like windows just binds to IPv4
https://github.com/hyperium/hyper/blob/e8d523e579a262e5901a00e6248bd315f0352afe/src/client/connect/http.rs#L592

https://stackoverflow.com/questions/12603096/how-to-make-a-tcp-socket-work-with-so-bindtodevice-against-routing-table

**Comment by Arriven** [comment]
Yeah, I'm not looking to implementing that on windows, just wanted to ensure that it works properly on unix without BindToDevice
